docs/standard/io/file-path-formats.md

```diff
@@ -0,0 +1,171 @@
+
+The DOS device path consists of the following components:
+
+- `\\.\` or `\\?\`, which identifies the path as a DOS device path.
+- A symbolic link to the "real" device object (C: in this case). 
+
+   There is a specific link for UNCs that is called, not surprisingly, `UNC`. For example:
+
+      \\.\UNC\Server\Share\Test\Foo.txt   
+      \\?\UNC\Server\Share\Test\Foo.txt
+
+Like UNCs, DOS device paths are fully qualified by definition. Current directories never enter into their usage.
+
+## Example: Ways to refer to the same file
+
+The following example illustrates some of the ways in which you can refer to a file when using the APIs in the <xref:System.IO> namespace. The example instantiates a <xref:System.IO.FileInfo> object and uses its <xref:System.IO.FileInfo.Name> and <xref:System.IO.FileInfo.Length> properties to display the filename and the length of the file.
+
+[!code-csharp[referring-to-the-same-file](~/samples/snippets/standard/io/file-names/cs/file-refs.cs)]
+[!code-vb[referring-to-the-same-file](~/samples/snippets/standard/io/file-names/vb/file-refs.vb)]
+
+## Path normalization
+
+Almost all paths passed to Windows APIs are normalized. During normalization, Windows performs the following steps:
+
+- Identifies the path.
+- Applies the current directory to partially qualified (relative) paths.
+- Canonicalizes component and directory separators.
+- Evaluates relative directory components (`.` for the current directory and `..` for the parent directory).
+- Trims certain characters.
+
+This normalization happens implicitly, but you can do it explicitly by calling the Windows [GetFullPathName() function](https://msdn.microsoft.com/library/windows/desktop/aa364963(v=vs.85).aspx).
+
+### Identifying the path
+
+The first step in path normalization is identifying the type of path. Paths fall into one of a few categories:
+
+- They are device paths; that is, they begin with two separators and a question mark or period (`\\?,` or `\\.`).
+- They are UNC paths; that is, they begin with two separators without a question mark or period. 
+- They are fully qualified DOS paths; that is, they begin with a drive letter, a volume separator, and a component separator (`C:\`).
+- They designate a legacy device (`CON`, `LPT1`).
+- They are relative to the root of the current drive; that is, they begin with a single component separator (`\`).
+- They are relative to the current directory of a specified drive; that is, they begin with a drive letter, a volume separator, and no component separator (`C:`).
+- They are relative to the current directory; that is, they begin with anything else (`temp\testfile.txt`).
+
+The type of the path determines whether or not a current directory is applied in some way. It also determines what the "root" of the path is.
+
+### Handling legacy devices
+
+If the path is a legacy DOS device such as `CON`, `COM1`, or `LPT1`, it is converted into a device path by prepending `\\.\` and returned.
```

Comment:
Note that `CON.TXT` comes back as `\\.\CON` as well. Any of these as is, or followed by a period. Comes up on StackOverflow.